### ISSUE TITLE
JSON field do not survive import & export.

### ISSUE THREAD

**Issue by BSVogler** [django-import-export/django-import-export#1216]
**Describe the bug**
JSON fields do not survive import/export.

**To Reproduce**
1. Have a model with a JSON field containing some JSON e.g. `{"a":"b"}`
2. Export using format CSV, JSON or TSV.
3. Open up new environment.
4. Import previous file
5. the json field contains the JSON in quotations and single quotes `"{'a':'b'}"`

**Versions (please complete the following information):**
 - Django Import Export: 2.4.0
 - Python 3.9
 - Django 3.1.2

**Expected behavior**
The field should have the same content as before.

**Comment by BSVogler** [comment]
The field was filled with a dictionary before and after import, it is a JSONstring. A `field = json.loads(field)` repairs it.
This might be only possible with one file format which escapes the quotes. File formoat which replace the double quotes with single ones can not be repaired this way.

**Comment by aureliendebord** [comment]
Same problem here. @BSVogler did you find a workaround for this?

**Comment by BSVogler** [comment]
@aureliendebord 

I call this method when I run into a type error when using this field post-import.

```python
def repair_json(djangoobject, field: str):
    """
    Repairs broken import/export strings.
    @param djangoobject: the broken django object, has side effect, as it might be repaired
    @param field: name of the field to repair
    @return:
    """
    # check if there is a chance that it is broken
    if hasattr(djangoobject, field) and (
            type(getattr(djangoobject, field)) == JSONString or type(getattr(djangoobject, field)) == str):
        jsonstring = getattr(djangoobject, field)
        jsonstring = jsonstring.removeprefix("\"")
        jsonstring = jsonstring.removesuffix("\"")
        jsonstring = jsonstring.replace("\'", "\"")
        jsonstring = jsonstring.replace("\\\"", "\"")
        jsonstring = jsonstring.replace("False", "false")
        jsonstring = jsonstring.replace("True", "true")
        setattr(djangoobject, field, json.loads(jsonstring))
        # even if validation will fail, this was improved
        djangoobject.save()
```

**Comment by matthewhegarty** [comment]
I had a shot at trying to reproduce this, but the import / export preserved the data dict OK for me.  Perhaps I'm missing something.

I've updated the example app on a test branch - @BSVogler perhaps you could try to reproduce following these steps?

```
git clone git@github.com:matthewhegarty/django-import-export.git
cd django-import-export
git checkout issue-1216

mkvirtualenv dj-imp-exp
pip install -r requirements/base.txt -r requirements/test.txt

cd tests

export IMPORT_EXPORT_TEST_TYPE=postgres
export IMPORT_EXPORT_POSTGRESQL_USER=pguser
export IMPORT_EXPORT_POSTGRESQL_PASSWORD=pguserpass
export DJANGO_SETTINGS_MODULE=settings

docker-compose -f bulk/docker-compose.yml up -d db

./manage.py migrate

# create superuser so that we can login to console
./manage.py createsuperuser --username admin  --email=email@example.com

./manage.py runserver
```

- login with admin / `<superuser pass>`
- Navigate to 'Books with chapterss'
- Click import, and import the attached file (BooksWithChapters-2021-04-30.csv)
- Open the imported object and you should see the 'data' field:

```
{"some_key": "some_value"}
```

- You can try an additional export / import to verify it imports OK.
  - increment the id field in the csv to create a new object

```
# Remove the db
docker-compose -f bulk/docker-compose.yml down -v
rmvirtualenv dj-imp-exp
```

**Comment by takkaria** [comment]
I can confirm that this example works, but I have an example that doesn't work.  I will try to boil down a test case.

**Comment by takkaria** [comment]
My story is:

* using django-import-export 2.5.0, exporting a load of rows with JSONFields as JSON led to them being escaped strangely.  I got data like this:
```
[
  {
    "json_row": "{'key': {'tag': 'xxx', 'cost': 400, 'tags': [None]}}"
  }
]
```

This looks like a serialized Python dict.

If I just upgrade my package version to django-import-export 2.6.1, I get something that looks like actual JSON when I export those rows:
```
[
  {
    "json_row": "{\"key\": {\"tag\": \"xxx\", \"cost\": 400, \"tags\": [null]}}"
  }
]
```

I didn't see anything in the changelog that helped me understand why this behaviour changed so I looked through the diff between 2.5.0 and 2.6.1 and the only thing I can find is the switch to using Django's native JSONField rather than the Postgres-specific JSONField in f348ef3782e60d44a2cedd0062179f0b9c798510.  So I think the new JSONField has some different serialisation behaviour to the old one.

In any case, I think this issue can be safely closed!  I upgraded the version of django-import-export we're using and the issue is fixed.
